inetutils: 1.9.4 -> 2.0

wahjava opened the pull request:
- Remove the already upstream-ed (or not applicable anymore) patches
 - Add patches to update .ORG, and .IN WHOIS servers (submitted upstream)

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

While trying to fix problem related to querying WHOIS server, I noticed available update


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

P. S. If maintainer prefers, I'll be happy to help with co-maintaining this derivation.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 120576` at a8aab896 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>libvirt_5_9_0</li>
    <li>linuxPackages-libre.zfs</li>
    <li>linuxPackages-libre.zfsStable</li>
    <li>linuxPackages-libre.zfsUnstable</li>
    <li>linuxPackages_hardkernel_4_14.zfs</li>
    <li>linuxPackages_hardkernel_4_14.zfsStable</li>
    <li>linuxPackages_hardkernel_4_14.zfsUnstable</li>
    <li>linuxPackages_hardkernel_latest.zfs</li>
    <li>linuxPackages_hardkernel_latest.zfsStable</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/8f1a164ca0c8e1ecf19e5e8015377001">libvmi</a></li>
    <li><a href="https://gist.github.com/0802d4bfacbb6bde993bfa96fff5ae88">qubes-core-vchan-xen</a></li>
    <li><a href="https://gist.github.com/8b1d82ab45b2e87c3e48dea621fc6394">rubyPackages_3_0.ruby-libvirt</a></li>
    <li><a href="https://gist.github.com/edf474ab65ac343654c46cd78390614c">xen</a> (xenPackages.xen-vanilla ,xenPackages.xen_4_10-vanilla ,xen_4_10)</li>
  </ul>
</details>
<details>
  <summary>44 packages skipped due to time constraints:</summary>
  <ul>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>gnome3.gnome-boxes</li>
    <li>libceph</li>
    <li>linuxPackages.zfs (linuxPackages.zfsStable ,linuxPackages_5_10.zfs ,linuxPackages_5_10.zfsStable)</li>
    <li>linuxPackages.zfsUnstable (linuxPackages_5_10.zfsUnstable)</li>
    <li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsStable)</li>
    <li>linuxPackages_4_14.zfsUnstable</li>
    <li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsStable)</li>
    <li>linuxPackages_4_19.zfsUnstable</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>42 packages built successfully:</summary>
  <ul>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>dnstwist</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>gnome3.gnome-nettool</li>
    <li>grml-zsh-config</li>
    <li>gsmartcontrol</li>
    <li>haskellPackages.libvirt-hs</li>
    <li>haskellPackages.libzfs</li>
    <li>inetutils</li>
    <li>libguestfs</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>linuxPackages_xen_dom0.zfsUnstable</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>nixopsUnstable</li>
    <li>partition-manager</li>
    <li>perl530Packages.SysVirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python38Packages.whois</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>python39Packages.whois</li>
    <li>rubyPackages.ruby-libvirt (rubyPackages_2_6.ruby-libvirt)</li>
    <li>rubyPackages_2_7.ruby-libvirt</li>
    <li>sanoid</li>
    <li>smartmontools</li>
    <li>terraform-providers.libvirt</li>
    <li>tlp</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-top</li>
    <li>virt-viewer</li>
    <li>virtlyst</li>
    <li>zfs (zfsStable)</li>
    <li>zfsUnstable</li>
    <li>zfstools</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

lukegb commented on pkgs/tools/networking/inetutils/default.nix:
```diff
@@ -1,18 +1,18 @@
 { stdenv, lib, fetchurl, ncurses, perl, help2man }:
 
 stdenv.mkDerivation rec {
-  name = "inetutils-1.9.4";
```
Please split to pname and version.

wahjava replied:
Done.

lukegb commented on pkgs/tools/networking/inetutils/default.nix:
```diff
@@ -1,18 +1,18 @@
+  name = "inetutils-2.0";
 
   src = fetchurl {
     url = "mirror://gnu/inetutils/${name}.tar.gz";
```
Is it worth pulling the xz variant of the tarball instead?

wahjava replied:
It's ~1M smaller, switched to it.

wahjava committed:
inetutils: 1.9.4 -> 2.0

- Remove the already upstream-ed (or not applicable anymore) patches
- Add patches to update .ORG, and .IN WHOIS servers (submitted upstream)
- Split name into pname, and version
- Switch to .xz distfile (smaller)

risicle commented:
`nixpkgs-review` happy on nixos x86_64, only build failures pre-existing. Ran various network-related nixos tests successfully too.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 120576` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
16 packages marked as broken and skipped:
</summary>
<ul>
<li>bareos</li>
<li>libvirt_5_9_0</li>
<li>linuxPackages-libre.zfs</li>
<li>linuxPackages-libre.zfsStable</li>
<li>linuxPackages-libre.zfsUnstable</li>
<li>linuxPackages_hardkernel_4_14.zfs</li>
<li>linuxPackages_hardkernel_4_14.zfsStable</li>
<li>linuxPackages_hardkernel_4_14.zfsUnstable</li>
<li>linuxPackages_hardkernel_latest.zfs</li>
<li>linuxPackages_hardkernel_latest.zfsStable</li>
<li>linuxPackages_hardkernel_latest.zfsUnstable</li>
<li>linuxPackages_latest-libre.zfs</li>
<li>linuxPackages_latest-libre.zfsStable</li>
<li>linuxPackages_latest-libre.zfsUnstable</li>
<li>samba4Full</li>
<li>sambaFull</li>
</ul>
</details>
<details>

  <summary>8 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>libvmi: log was empty</li>
<li>qemu_xen: log was empty</li>
<li>qemu_xen-light: log was empty</li>
<li>qubes-core-vchan-xen: log was empty</li>
<li>rubyPackages_3_0.ruby-libvirt: log was empty</li>
<li>xen: log was empty</li>
<li>xen-light: log was empty</li>
<li>xen-slim: log was empty</li>
  </ul>
</details>
<details>

  <summary>11 packages failed to build and are new build failures:</summary>
  <ul>
<li>(qemu_xen_4_10): log was empty</li>
<li>(qemu_xen_4_10-light): log was empty</li>
<li>(xenPackages.xen-vanilla: log was empty</li>
<li>,xenPackages.xen_4_10-vanilla: log was empty</li>
<li>,xen_4_10): log was empty</li>
<li>(xenPackages.xen-light: log was empty</li>
<li>,xenPackages.xen_4_10-light: log was empty</li>
<li>,xen_4_10-light): log was empty</li>
<li>(xenPackages.xen-slim: log was empty</li>
<li>,xenPackages.xen_4_10-slim: log was empty</li>
<li>,xen_4_10-slim): log was empty</li>
  </ul>
</details>
<details>
<summary>
74 packages built:
</summary>
<ul>
<li>ceph (ceph-dev)</li>
<li>ceph-client</li>
<li>collectd</li>
<li>collectd-data</li>
<li>dnstwist</li>
<li>docker-machine-kvm</li>
<li>docker-machine-kvm2</li>
<li>easysnap</li>
<li>gnome3.gnome-boxes</li>
<li>gnome3.gnome-nettool</li>
<li>grml-zsh-config</li>
<li>gsmartcontrol</li>
<li>haskellPackages.libvirt-hs</li>
<li>haskellPackages.libzfs</li>
<li>inetutils</li>
<li>libceph</li>
<li>libguestfs</li>
<li>libvirt</li>
<li>libvirt-glib</li>
<li>linuxPackages.zfs (linuxPackages.zfsStable ,linuxPackages_5_10.zfs ,linuxPackages_5_10.zfsStable ,linuxPackages_xen_dom0.zfs ,linuxPackages_xen_dom0.zfsStable)</li>
<li>linuxPackages.zfsUnstable (linuxPackages_5_10.zfsUnstable ,linuxPackages_xen_dom0.zfsUnstable)</li>
<li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsStable)</li>
<li>linuxPackages_4_14.zfsUnstable</li>
<li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsStable)</li>
<li>linuxPackages_4_19.zfsUnstable</li>
<li>linuxPackages_4_4.zfs (linuxPackages_4_4.zfsStable)</li>
<li>linuxPackages_4_4.zfsUnstable</li>
<li>linuxPackages_4_9.zfs (linuxPackages_4_9.zfsStable)</li>
<li>linuxPackages_4_9.zfsUnstable</li>
<li>linuxPackages_5_11.zfs (linuxPackages_5_11.zfsStable ,linuxPackages_latest.zfs ,linuxPackages_latest.zfsStable ,linuxPackages_latest_xen_dom0.zfs ,linuxPackages_latest_xen_dom0.zfsStable)</li>
<li>linuxPackages_5_11.zfsUnstable (linuxPackages_latest.zfsUnstable ,linuxPackages_latest_xen_dom0.zfsUnstable)</li>
<li>linuxPackages_5_4.zfs (linuxPackages_5_4.zfsStable)</li>
<li>linuxPackages_5_4.zfsUnstable</li>
<li>linuxPackages_hardened.zfs (linuxPackages_hardened.zfsStable ,linuxPackages_xen_dom0_hardened.zfs ,linuxPackages_xen_dom0_hardened.zfsStable)</li>
<li>linuxPackages_hardened.zfsUnstable (linuxPackages_xen_dom0_hardened.zfsUnstable)</li>
<li>linuxPackages_latest_hardened.zfs (linuxPackages_latest_hardened.zfsStable ,linuxPackages_latest_xen_dom0_hardened.zfs ,linuxPackages_latest_xen_dom0_hardened.zfsStable)</li>
<li>linuxPackages_latest_hardened.zfsUnstable (linuxPackages_latest_xen_dom0_hardened.zfsUnstable)</li>
<li>linuxPackages_lqx.zfs (linuxPackages_lqx.zfsStable)</li>
<li>linuxPackages_lqx.zfsUnstable</li>
<li>linuxPackages_testing_bcachefs.zfs (linuxPackages_testing_bcachefs.zfsStable)</li>
<li>linuxPackages_testing_bcachefs.zfsUnstable</li>
<li>linuxPackages_xanmod.zfs (linuxPackages_xanmod.zfsStable)</li>
<li>linuxPackages_xanmod.zfsUnstable</li>
<li>linuxPackages_zen.zfs (linuxPackages_zen.zfsStable)</li>
<li>linuxPackages_zen.zfsUnstable</li>
<li>minikube</li>
<li>minishift</li>
<li>nixopsUnstable</li>
<li>partition-manager</li>
<li>perl530Packages.SysVirt</li>
<li>perl532Packages.SysVirt</li>
<li>python38Packages.guestfs</li>
<li>python38Packages.libvirt</li>
<li>python38Packages.whois</li>
<li>python39Packages.guestfs</li>
<li>python39Packages.libvirt</li>
<li>python39Packages.whois</li>
<li>qemu_full</li>
<li>rubyPackages.ruby-libvirt (rubyPackages_2_6.ruby-libvirt)</li>
<li>rubyPackages_2_7.ruby-libvirt</li>
<li>sanoid</li>
<li>smartmontools</li>
<li>terraform-full</li>
<li>terraform-providers.libvirt</li>
<li>tlp</li>
<li>vagrant</li>
<li>virt-manager</li>
<li>virt-manager-qt</li>
<li>virt-top</li>
<li>virt-viewer</li>
<li>virtlyst</li>
<li>zfs (zfsStable)</li>
<li>zfsUnstable</li>
<li>zfstools</li>
</ul>
</details>

tomberek commented:
Result of `nixpkgs-review pr 120576` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>libvirt_5_9_0</li>
    <li>linuxPackages-libre.zfs</li>
    <li>linuxPackages-libre.zfsStable</li>
    <li>linuxPackages-libre.zfsUnstable</li>
    <li>linuxPackages_hardkernel_4_14.zfs</li>
    <li>linuxPackages_hardkernel_4_14.zfsStable</li>
    <li>linuxPackages_hardkernel_4_14.zfsUnstable</li>
    <li>linuxPackages_hardkernel_latest.zfs</li>
    <li>linuxPackages_hardkernel_latest.zfsStable</li>
    <li>linuxPackages_hardkernel_latest.zfsUnstable</li>
    <li>linuxPackages_latest-libre.zfs</li>
    <li>linuxPackages_latest-libre.zfsStable</li>
    <li>linuxPackages_latest-libre.zfsUnstable</li>
    <li>samba4Full</li>
    <li>sambaFull</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>libvmi</li>
    <li>qemu_xen (qemu_xen_4_10)</li>
    <li>qemu_xen-light (qemu_xen_4_10-light)</li>
    <li>qubes-core-vchan-xen</li>
    <li>rubyPackages_3_0.ruby-libvirt</li>
    <li>xen (xenPackages.xen-vanilla ,xenPackages.xen_4_10-vanilla ,xen_4_10)</li>
    <li>xen-light (xenPackages.xen-light ,xenPackages.xen_4_10-light ,xen_4_10-light)</li>
    <li>xen-slim (xenPackages.xen-slim ,xenPackages.xen_4_10-slim ,xen_4_10-slim)</li>
  </ul>
</details>
<details>
  <summary>74 packages built:</summary>
  <ul>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>dnstwist</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>easysnap</li>
    <li>gnome3.gnome-boxes</li>
    <li>gnome3.gnome-nettool</li>
    <li>grml-zsh-config</li>
    <li>gsmartcontrol</li>
    <li>haskellPackages.libvirt-hs</li>
    <li>haskellPackages.libzfs</li>
    <li>inetutils</li>
    <li>libceph</li>
    <li>libguestfs</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>linuxPackages.zfs (linuxPackages.zfsStable ,linuxPackages_5_10.zfs ,linuxPackages_5_10.zfsStable ,linuxPackages_xen_dom0.zfs ,linuxPackages_xen_dom0.zfsStable)</li>
    <li>linuxPackages.zfsUnstable (linuxPackages_5_10.zfsUnstable ,linuxPackages_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_4_14.zfs (linuxPackages_4_14.zfsStable)</li>
    <li>linuxPackages_4_14.zfsUnstable</li>
    <li>linuxPackages_4_19.zfs (linuxPackages_4_19.zfsStable)</li>
    <li>linuxPackages_4_19.zfsUnstable</li>
    <li>linuxPackages_4_4.zfs (linuxPackages_4_4.zfsStable)</li>
    <li>linuxPackages_4_4.zfsUnstable</li>
    <li>linuxPackages_4_9.zfs (linuxPackages_4_9.zfsStable)</li>
    <li>linuxPackages_4_9.zfsUnstable</li>
    <li>linuxPackages_5_11.zfs (linuxPackages_5_11.zfsStable ,linuxPackages_latest.zfs ,linuxPackages_latest.zfsStable ,linuxPackages_latest_xen_dom0.zfs ,linuxPackages_latest_xen_dom0.zfsStable)</li>
    <li>linuxPackages_5_11.zfsUnstable (linuxPackages_latest.zfsUnstable ,linuxPackages_latest_xen_dom0.zfsUnstable)</li>
    <li>linuxPackages_5_4.zfs (linuxPackages_5_4.zfsStable)</li>
    <li>linuxPackages_5_4.zfsUnstable</li>
    <li>linuxPackages_hardened.zfs (linuxPackages_hardened.zfsStable ,linuxPackages_xen_dom0_hardened.zfs ,linuxPackages_xen_dom0_hardened.zfsStable)</li>
    <li>linuxPackages_hardened.zfsUnstable (linuxPackages_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_latest_hardened.zfs (linuxPackages_latest_hardened.zfsStable ,linuxPackages_latest_xen_dom0_hardened.zfs ,linuxPackages_latest_xen_dom0_hardened.zfsStable)</li>
    <li>linuxPackages_latest_hardened.zfsUnstable (linuxPackages_latest_xen_dom0_hardened.zfsUnstable)</li>
    <li>linuxPackages_lqx.zfs (linuxPackages_lqx.zfsStable)</li>
    <li>linuxPackages_lqx.zfsUnstable</li>
    <li>linuxPackages_testing_bcachefs.zfs (linuxPackages_testing_bcachefs.zfsStable)</li>
    <li>linuxPackages_testing_bcachefs.zfsUnstable</li>
    <li>linuxPackages_xanmod.zfs (linuxPackages_xanmod.zfsStable)</li>
    <li>linuxPackages_xanmod.zfsUnstable</li>
    <li>linuxPackages_zen.zfs (linuxPackages_zen.zfsStable)</li>
    <li>linuxPackages_zen.zfsUnstable</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>nixopsUnstable</li>
    <li>partition-manager</li>
    <li>perl530Packages.SysVirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python38Packages.whois</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>python39Packages.whois</li>
    <li>qemu_full</li>
    <li>rubyPackages.ruby-libvirt (rubyPackages_2_6.ruby-libvirt)</li>
    <li>rubyPackages_2_7.ruby-libvirt</li>
    <li>sanoid</li>
    <li>smartmontools</li>
    <li>terraform-full</li>
    <li>terraform-providers.libvirt</li>
    <li>tlp</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-manager-qt</li>
    <li>virt-top</li>
    <li>virt-viewer</li>
    <li>virtlyst</li>
    <li>zfs (zfsStable)</li>
    <li>zfsUnstable</li>
    <li>zfstools</li>
  </ul>
</details>

tomberek commented:
Should this go to staging?

davidak reviewed (approved):
Change looks good
